### DISCUSSION TITLE
Why does the `hugo` formulae not install `go` ?

### DISCUSSION THREAD

**Discussion by asmaier** [Homebrew/discussions#4970] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: ab6a7ed5a122b6d4f9d441ef76c86032398ced5f
Last commit: 4 days ago
Core tap JSON: 09 Dec 11:07 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 14.0.0 build 1400
Git: 2.41.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 12.7.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libDellMonitorSdkLib.dylib
```

### Description of issue

I see that the `hugo` formula depends on `go`, see https://github.com/Homebrew/homebrew-core/blob/7ce587a39e9968d73feaa59b779fbe6b0da99003/Formula/h/hugo.rb#L20 . However after 

    brew install hugo
    
the formula for `go` does not get installed. So the command `go` is not available and `hugo server` can throw an error like 

    Error: command error: failed to load modules: failed to download modules: binary with name "go" not found

That is why installation guides for `hugo`like https://docs.hugoblox.com/getting-started/install-hugo/ need to install the `golang` dependency manually. But why is this necessary? Why doesn't `go` get installed automatically, if it is a dependency? Is that a bug?

**Comment by SMillerDev** [comment]
It only needs Go to compile the binary as far as the formula defines it, so for normal usage it won't install Go. The test also passes without issues so clearly some parts work without Go.

I'd happily review a pull request that extends the test to indicate a Go dependency though.
